src/Controls/src/SourceGen/CodeBehindGenerator.cs

```diff
@@ -137,7 +143,12 @@ static AssemblyCaches GetAssemblyAttributes(Compilation compilation, Cancellatio
 			return new AssemblyCaches(xmlnsDefinitions, internalsVisible);
 		}
 
-		static void GenerateXamlCodeBehind(ProjectItem projItem, Compilation compilation, SourceProductionContext context, AssemblyCaches caches)
+		static IDictionary<XmlType, string> GetTypeCache(Compilation compilation, CancellationToken cancellationToken)
+		{
+			return new Dictionary<XmlType, string>();
```

Comment:
`GetHashCode()` is implemented on `XmlType`, so this seems like it would OK to use as a key in a dictionary.

Reply:
Hashcode values aren't necessarily unique. They just help with bucketization. If we had huge numbers of types that only differ in type arguments then it would be worth it, but here it should be fine.